src/DatasetBrowser/index.jsx

```diff
@@ -0,0 +1,270 @@
+import React from 'react';
+import FilterGroup from '@gen3/ui-component/dist/components/filters/FilterGroup';
+import FilterList from '@gen3/ui-component/dist/components/filters/FilterList';
+import { config } from '../params';
+import DatasetBrowserTable from './DatasetBrowserTable/';
+import DataSummaryCardGroup from '../components/cards/DataSummaryCardGroup/.';
+import './DatasetBrowser.less';
+import { fetchWithCreds } from '../actions';
+import { guppyGraphQLUrl } from '../configs';
+
+function calculateSummaryCounts(field, filteredData) {
+  const values = [];
+  for(let j = 0; j < filteredData.length; j+= 1) {
+    values.push(filteredData[j][field]);
+  }
+  const uniqueValues = values.filter(
+    (value, index) => values.indexOf(value) === index 
+  );
+  return uniqueValues.length;
+}
+
+function checkIfFiltersApply(filtersApplied, row) {
+  const attributes = Object.keys(filtersApplied);
+  for (let i = 0; i < attributes.length; i += 1) {
+    const property = attributes[i];
+    if (!row[property]) {
+      return false;
+    }
+    const filtersApplyMatch = filtersApplied[property].selectedValues.map(
+      x => x.toLowerCase()
+    ).includes(
+      row[property].toLowerCase()
+    );
+    let filtersApplyContains = filtersApplied[property].selectedValues.filter(
+      x => row[property].toLowerCase().includes(x.toLowerCase())
+    );
+    filtersApplyContains = filtersApplyContains.length > 0;
+    const filtersApply = filtersApplyMatch || filtersApplyContains;
+    if (!filtersApply) {
+      return false;
+    }
+  }
+  return true;
+}
+
+class DatasetBrowser extends React.Component {
+  constructor(props) {
+    super(props);
+    const tabIndex = 0;
+    this.state = {
+      tab: tabIndex > 0 ? tabIndex : 0,
+      rawData: [],
+      filteredData: [],
+      paginatedData: [],
+      counts: {
+        'supported_data_resource': 0,
+        'dataset_name': 0
+      }
+    };
+    this.filterGroupRef = React.createRef();
+    this.tableRef = React.createRef();
+  }
+
+  componentWillMount() {
+    this.initializeData();
+  }
+
+  obtainSubcommonsData = (subcommonsConfig) => {
+    const subcommonsURL = subcommonsConfig.URL;
+    const subcommonsName = subcommonsConfig.name;
+    const graphModelQueryURL = 'api/v0/submission/graphql';
+    const queryString = `
+      {
+        study {
+          study_description
+          submitter_id
+          study_design
+        }
+      }
+    `;
+    return fetch(subcommonsURL + graphModelQueryURL, {
+      method: 'POST',
+      body: JSON.stringify({
+        query: queryString,
+      })
+    }).then(
+      result => result.json(), 
+      reason => [ ] // eslint-disable-line
```

Comment:
is this line needed?

Reply:
Yes! If the async fetch to a commons fails, this ensures that there will at least be an empty array delivered to the next callback. Without this line, the user will receive no data from the other subcommons if one of the subcommons gives them a 401.